### PR TITLE
Add JDK libraries in a more flexible way based off java home

### LINKED ISSUE #25: assemble fails in task jsonp.1.0:jar: Files do not exist
Attempting to follow the contributing instructions on macOS 10.12. Cloning and `cnf:initialize` work fine, but `assemble` or `releaseNeeded` fail with `Project com.ibm.websphere.javaee.jsonp.1.0 has errors, 24 errors were reported`. I note that the files all have names starting with "ibm" and that it is looking at the JRE on my `$PATH` which is an Oracle JRE.

Full output:

```
PowerBook-i73000:dev liamwestby$ ./gradlew --stacktrace assemble

> Configure project :com.ibm.websphere.javaee.management.j2ee.1.1
-d /Users/liamwestby/Documents/Programming/open-liberty/dev/com.ibm.websphere.javaee.management.j2ee.1.1/build/classes/java/main
-cp /Users/liamwestby/Documents/Programming/open-liberty/dev/com.ibm.websphere.javaee.management.j2ee.1.1/build/classes/java/main:/Users/liamwestby/Documents/Programming/open-liberty/dev/com.ibm.websphere.javaee.ejb.3.2/build/libs/com.ibm.websphere.javaee.ejb.3.2.jar:/Users/liamwestby/Documents/Programming/open-liberty/dev/com.ibm.ws.org.osgi.annotation.versioning/build/libs/com.ibm.ws.org.osgi.annotation.versioning.jar

> Task :com.ibm.websphere.javaee.jsonp.1.0:jar
Warning: There is an empty jar or directory on the classpath: main
Warning: Imports that lack version ranges: [org.glassfish.json]
Warning: Unreferenced class path entries [javax.json-1.0.4, main]
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmcertpathfw.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjcefw.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjgssfw.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjgssprovider.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjssefw.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmorbapi.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmpkcs.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/math.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/rt.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/bin/default/jclSC180/vm.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/xml.jar
Error  : Cannot find /error/File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/xmldsigfw.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmcertpathfw.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjcefw.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjgssfw.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjgssprovider.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmjssefw.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmorbapi.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/ibmpkcs.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/math.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/rt.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/bin/default/jclSC180/vm.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/xml.jar
Error  : File does not exist: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/jre/lib/xmldsigfw.jar


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':com.ibm.websphere.javaee.jsonp.1.0:jar'.
> Project com.ibm.websphere.javaee.jsonp.1.0 has errors, 24 errors were reported

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':com.ibm.websphere.javaee.jsonp.1.0:jar'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:242)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:95)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:235)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:224)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:121)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:77)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:102)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:96)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:612)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:567)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:96)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.api.GradleException: Project com.ibm.websphere.javaee.jsonp.1.0 has errors, 24 errors were reported
        at aQute.bnd.gradle.BndPlugin.checkErrors(BndPlugin.groovy:515)
        at aQute.bnd.gradle.BndPlugin.checkErrors(BndPlugin.groovy)
        at aQute.bnd.gradle.BndPlugin.this$2$checkErrors(BndPlugin.groovy)
        at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:479)
        at org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:191)
        at org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:80)
        at org.gradle.internal.metaobject.BeanDynamicObject$GroovyObjectAdapter.invokeOpaqueMethod(BeanDynamicObject.java:579)
        at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:506)
        at org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:191)
        at org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:80)
        at aQute.bnd.gradle.BndPlugin$_apply_closure1_closure13_closure48.doCall(BndPlugin.groovy:264)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:670)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:651)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:95)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        ... 27 more


BUILD FAILED in 48s
5 actionable tasks: 3 executed, 2 up-to-date
```

### PR DESCRIPTION
**Pull request by ewhoch**
fixes #25 . Allows for a more flexible adding of JDK libraries from a user's java home directory.

### CODE REVIEW

File: dev/settings.gradle [hunk 1]
```diff
@@ -76,11 +76,34 @@ def loadBuildProps() {
...
   
+
+  if (isFirstPass) {
+     jdkLibString = jdkLibString + file + ";version=file;boot=true"
+  } else {
+     jdkLibString = jdkLibString + ", " + file + ";version=file;boot=true"
+  }
+	 isFirstPass = false;
+  }
+  println jdkLibString
```
Comment (aguibert) [inline dev/settings.gradle]: be sure to clean up this print statement before delivering

**Comment by ewhoch** [comment]
Updated the commit to remove a noted println

**Comment by realModusOperandi** [comment]
Confirming that this gets me past the file not found errors.

**Comment by ewhoch** [comment]
#build

**Comment by ewhoch** [comment]
Doing a final local regression test and then i'll merge this in.

**Pull request merged**